cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1040,16 +1046,62 @@ aggregate_reader_metadata::select_row_groups(
   rmm::cuda_stream_view stream) const
 {
   std::optional<std::vector<std::vector<size_type>>> filtered_row_group_indices;
+
+  // Compute total number of input row groups if needed
+  size_type total_row_groups = [&]() {
+    if (not row_group_indices.empty()) {
+      return std::accumulate(
+        row_group_indices.begin(),
+        row_group_indices.end(),
+        size_type{0},
+        [](auto sum, auto const& per_file_row_groups) { return sum + per_file_row_groups.size(); });
+    } else {
+      return num_row_groups;
+    }
+  }();
+
+  // Initialize to the total number of row groups
+  size_type num_stats_filtered_row_groups = total_row_groups;
+  size_type num_bloom_filtered_row_groups = total_row_groups;
+
   // if filter is not empty, then gather row groups to read after predicate pushdown
   if (filter.has_value()) {
-    filtered_row_group_indices = filter_row_groups(
-      sources, row_group_indices, output_dtypes, output_column_schemas, filter.value(), stream);
+    // Span of input row group indices for predicate pushdown
```

Comment:
Moved this piece of code out of `filter_row_groups()` to make its signature consistent with `apply_bloom_filters()`